{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "gpuType": "T4",
      "toc_visible": true,
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "GPU",
    "gpuClass": "standard"
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/paulynamagana/3D-Beacons/blob/main/Tutorials/Harnessing_3DBeaconsAPI.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "<h1>Accessing data though 3D-Beacons API</h1>\n",
        "\n",
        "<img src=\"https://raw.githubusercontent.com/3D-Beacons/3D-Beacons/main/assets/3D-Beacons-logo.png\" height=\"100\" align=\"right\">\n",
        "\n",
        "## Introduction\n",
        "\n",
        "Welcome to this Google Colab notebook, an essential companion to our paper on accessing data through the 3D-Beacons API.\n",
        "\n",
        "This notebook serves as a practical resource for researchers, developers, and data enthusiasts who are interested in exploring the potential of the 3D-Beacons API and leveraging its capabilities. By following this guide, you will gain a deep understanding of how to interact with the 3D-Beacons API to access and analyze various types of data.\n",
        "To supplement your learning, we have provided links to the full paper as well as documentation resources that will assist you in navigating the API effectively.\n",
        "\n",
        "Documentation Link: [3D-Beacons API Documentation](https://www.ebi.ac.uk/pdbe/pdbe-kb/3dbeacons/api/#/default/get_uniprot_summary_uniprot_summary__qualifier__json_get)\n",
        "\n",
        "<br>\n",
        "\n",
        "**Further readings:**\n",
        "\n",
        "**3D-Beacons: Decreasing the gap between protein sequences and structures through a federated network of protein structure data resources**\n",
        "\n",
        "Mihaly Varadi, Sreenath Nair, Ian Sillitoe, Gerardo Tauriello, Stephen Anyango, Stefan Bienert, Clemente Borges, Mandar Deshpande, Tim Green, Demis Hassabis, Andras Hatos, Tamas Hegedus, Maarten L Hekkelman, Robbie Joosten, John Jumper, Agata Laydon, Dmitry Molodenskiy, Damiano Piovesan, Edoardo Salladini, Steven L. Salzberg, Markus J Sommer, Martin Steinegger, Erzsebet Suhajda, Dmitri Svergun, Luiggi Tenorio-Ku, Silvio Tosatto, Kathryn Tunyasuvunakool, Andrew Mark Waterhouse, Augustin Žídek, Torsten Schwede, Christine Orengo, Sameer Velankar\n",
        "3 August 2022; BioRxiv https://doi.org/10.1101/2022.08.01.501973\n"
      ],
      "metadata": {
        "id": "NhWW96FJyM8n"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "  ## Instructions <a name=\"INSTRUCTIONS\"></a>\n",
        "\n",
        "* Quick Start <a name=\"Quick Start\"></a>\n",
        "\n",
        "In order to make the learning experience more accessible and interactive, we have incorporated widgets that allow you to provide inputs and customize certain aspects of the code. These widgets serve as interactive elements that enhance your ability to interact with the code and observe the impact of different inputs on the results. Throughout this tutorial, you will come across code chunks that generate these interactive widgets.\n",
        "\n",
        "  * How to use Google Colab <a name=\"Google Colab\"></a>\n",
        "1. Before running the code, make sure you have filled the necessary information in the input form.\n",
        "2. To run a code cell, click on the cell to select it. You will notice a play button (▶️) on the left side of the cell. Click on the play button or press Shift+Enter to run the code in the selected cell.\n",
        "3. The code will start executing, and you will see the output, if any, displayed below the code cell.\n",
        "4. Move to the next code cell and repeat steps 2 and 3 until you have executed all the desired code cells in sequence.\n",
        "5. The currently running step is indicated by a circle with a stop sign next to it.\n",
        "If you need to stop or interrupt the execution of a code cell, you can click on the stop button (■) located next to the play button.\n",
        "\n",
        "\n",
        "*Remember to run the code cells in the correct order, as their execution might depend on variables or functions defined in previous cells. You can modify the code in a code cell and re-run it to see updated results.*\n",
        "\n",
        "*Note: If the notebook runtime is restarted you will need to re-run the first 3 code chunks located in the Set Up section.*\n",
        "\n",
        "\n",
        "\n",
        "## Contact us\n",
        "\n",
        "Requests for clarifications or reporting bugs can be made via pdbekb_help@ebi.ac.uk.\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "j5tiopj9fZFJ"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "\n",
        "---\n",
        "\n"
      ],
      "metadata": {
        "id": "N-Tmi3KmZE3u"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {
        "id": "R7fad_36A4PL",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "32c1ed7d-2312-4625-9080-1fc1b05f0ee8",
        "cellView": "form"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Succesfully installed!\n",
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ]
        }
      ],
      "source": [
        "#@title Set Up: Install libraries and mount Google Drive\n",
        "\n",
        "#@markdown After running this code, you may be prompted to grant access to your Google Drive. This is necessary for Google Colab to download the files and save them to your Drive.\n",
        "#@markdown <br>\n",
        "#@markdown <br>\n",
        "#@markdown Please follow the on-screen instructions to provide the necessary permissions, as it enables seamless integration between Colab and your Drive. Rest assured that your data and files are secure and will not be accessed without your explicit permission.\n",
        "\n",
        "!pip install ijson gwpy &> /dev/null\n",
        "\n",
        "print(\"Succesfully installed!\")\n",
        "\n",
        "#@title Mount Drive to download files\n",
        "######## LIBRARIES\n",
        "import ijson\n",
        "import requests, sys, json\n",
        "import ipywidgets as wgt\n",
        "from urllib.request import urlopen\n",
        "# Importing the necessary libraries for mounting Google Drive\n",
        "import os\n",
        "from google.colab import drive\n",
        "\n",
        "\n",
        "######## FUNCTIONS\n",
        "#defining functions for search and download\n",
        "def Search3DBeacons(ID):\n",
        "  WEBSITE_API = \"https://www.ebi.ac.uk/pdbe/pdbe-kb/3dbeacons/api/v2/uniprot/summary/\"\n",
        "\n",
        "  r = ijson.parse(urlopen(f\"{WEBSITE_API}{ID}.json\"))\n",
        "  structures = list(ijson.items(r, \"structures.item\", use_float=True))\n",
        "  return structures\n",
        "\n",
        "\n",
        "# Defining function for sequence search\n",
        "def sequence_search(sequence):\n",
        "    global job_id\n",
        "\n",
        "    post_url = \"https://www.ebi.ac.uk/pdbe/pdbe-kb/3dbeacons/api/v2/sequence/search\"\n",
        "    query_sequence = {\"sequence\": sequence}\n",
        "\n",
        "    try:\n",
        "        response = requests.post(post_url, json=query_sequence)\n",
        "        response.raise_for_status()  # Raise an exception if it fails\n",
        "        if response.status_code == 200:\n",
        "            print(\"Your search was successful\")\n",
        "            job_id = response.json()[\"job_id\"]\n",
        "            retrieve_results(job_id)\n",
        "        else:\n",
        "            print(f\"Request failed with status code {response.status_code}\")\n",
        "            exit()\n",
        "    except requests.RequestException as e:\n",
        "        print(f\"Request failed with status code: {response.status_code}\")\n",
        "        print(f\"Response text: {response.text}\")\n",
        "        exit()\n",
        "\n",
        "def retrieve_results(job_id):\n",
        "    get_url = \"https://www.ebi.ac.uk/pdbe/pdbe-kb/3dbeacons/api/v2/sequence/result\"\n",
        "    try:\n",
        "        # Use requests.get for better error handling\n",
        "        response = requests.get(f\"{get_url}?job_id={job_id}\")\n",
        "        response.raise_for_status()\n",
        "\n",
        "        # Parse the JSON response with ijson\n",
        "        items = ijson.items(response.content, 'item')\n",
        "        for item in items:\n",
        "            print(item)\n",
        "    except requests.RequestException as e:\n",
        "        print(f\"An error occurred during the GET request: {e}\")\n",
        "        exit()\n",
        "\n",
        "\n",
        "# Defining function for Ensembl search\n",
        "def ensembl_search(ENSEMBL_ID):\n",
        "    url = \"https://www.ebi.ac.uk/pdbe/pdbe-kb/3dbeacons/api/v2/ensembl/summary/\" #url\n",
        "\n",
        "    try:\n",
        "        # Send GET request to retrieve Ensembl summary\n",
        "        response = urlopen(f\"{url}{ENSEMBL_ID}.json\")\n",
        "        if response.status == 200:\n",
        "            print(\"Ensembl search successful\")\n",
        "        else:\n",
        "            print(f\"Ensembl search failed with status code {response.status}\")\n",
        "\n",
        "        r = ijson.parse(response)\n",
        "        ensembls = ijson.items(r, \"uniprot_mappings.item\", use_float=True)\n",
        "\n",
        "        # Loop for better printing\n",
        "        for ensembl in ensembls:\n",
        "            print(ensembl)\n",
        "    except Exception as e:\n",
        "        print(f\"An error occurred during the Ensembl search: {e}\")\n",
        "        exit()\n",
        "\n",
        "######## API\n",
        "#url_API\n",
        "WEBSITE_API = \"https://www.ebi.ac.uk/pdbe/pdbe-kb/3dbeacons/api/v2/uniprot/summary/\"\n",
        "\n",
        "######## GOOGLE DRIVE\n",
        "# Mounting the Google Drive to access files and directories\n",
        "drive.mount('/content/drive')\n",
        "destination_path = '/content/drive/MyDrive/3D_Beacons_files'\n",
        "\n",
        "# Check whether the specified path exists or not\n",
        "isExist = os.path.exists(destination_path)\n",
        "if not isExist:\n",
        "   # Create a new directory because it does not exist\n",
        "   os.makedirs(destination_path)\n",
        "   print(\"The new directory is created!\")\n",
        "\n",
        "def download_file(model_url, destination_file_path):\n",
        "  try:\n",
        "    urllib.request.urlretrieve(model_url, destination_file_path)\n",
        "    print(f'File downloaded successfully to: {destination_file_path}')\n",
        "  except urllib.error.URLError as e:\n",
        "    print(f'Error downloading file: {e}')  # Handle URL-related errors\n",
        "  except Exception as e:  # Catch other general exceptions\n",
        "    print(f'An unexpected error occurred: {e}')\n"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "\n",
        "---\n",
        "\n"
      ],
      "metadata": {
        "id": "HBfiYKdZY8n5"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "##1.&nbsp; BASIC SEARCH\n",
        "\n",
        "This section implements API to search and filter predicted and experimentally determined structures available from the 3D-Beacons network.\n",
        "\n",
        "Once you have found a model, you can download the specific"
      ],
      "metadata": {
        "id": "LlKjS1E9Q_rS"
      }
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "Mmb_qCzNJMNG"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 1.1.&nbsp;  Get all macromolecular structures for a single entry\n",
        "#@markdown  On this sub-section you will be able to retrieve all available structures and metainformation in 3D-Beacons from a single Uniprot accession ID.\n",
        "#Run this code to display the widget\n",
        "Uniprot_ID = \"P00520\" #@param {type:\"string\"}\n",
        "\n",
        "#Run this code to search the network using the input from the widget\n",
        "structures = Search3DBeacons(Uniprot_ID)\n",
        "\n",
        "#print available structures from different providers\n",
        "for structure in structures:\n",
        "    print(structure)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Vqp8OHciRewN",
        "outputId": "07d290b8-6d88-4776-9ced-bca12eaffafb",
        "cellView": "form"
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "{'summary': {'model_identifier': '3kfa', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3kfa_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3kfa', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-10-27', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.22'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'B91', 'identifier_category': 'CCD', 'description': '3-{(E)-2-[6-(cyclopropylamino)-9H-purin-9-yl]ethenyl}-4-methyl-N-[3-(4-methyl-1H-imidazol-1-yl)-5-(trifluoromethyl)phenyl]benzamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3k5v', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3k5v_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3k5v', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-10-08', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.74'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STJ', 'identifier_category': 'CCD', 'description': '3-(6-{[4-(trifluoromethoxy)phenyl]amino}pyrimidin-4-yl)benzamide', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1opj', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1opj_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1opj', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2003-03-06', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.75'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MYR', 'identifier_category': 'CCD', 'description': 'MYRISTIC ACID', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1opk', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1opk_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1opk', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2003-03-06', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 27, 'uniprot_end': 515, 'coverage': Decimal('0.435'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.8'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'GOL', 'identifier_category': 'CCD', 'description': 'GLYCEROL', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MYR', 'identifier_category': 'CCD', 'description': 'MYRISTIC ACID', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'P16', 'identifier_category': 'CCD', 'description': '6-(2,6-DICHLOROPHENYL)-2-{[3-(HYDROXYMETHYL)PHENYL]AMINO}-8-METHYLPYRIDO[2,3-D]PYRIMIDIN-7(8H)-ONE', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '3ms9', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3ms9_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3ms9', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2010-04-29', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.8'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MS9', 'identifier_category': 'CCD', 'description': 'methyl 2-amino-4-chlorobenzoate', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '5ih2', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/5ih2_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/5ih2', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2016-02-28', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 757, 'uniprot_end': 765, 'coverage': Decimal('0.008'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.8'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proline rich Peptide', 'chain_ids': ['M', 'N']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'Q64010', 'identifier_category': 'UNIPROT', 'description': 'Adapter molecule crk', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'NA', 'identifier_category': 'CCD', 'description': 'SODIUM ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'PEG', 'identifier_category': 'CCD', 'description': 'DI(HYDROXYETHYL)ETHER', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'P4G', 'identifier_category': 'CCD', 'description': '1-ETHOXY-2-(2-ETHOXYETHOXY)ETHANE', 'chain_ids': ['B', 'N']}]}}\n",
            "{'summary': {'model_identifier': '3kf4', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3kf4_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3kf4', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-10-27', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.9'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'B90', 'identifier_category': 'CCD', 'description': 'N-[4-(dimethylphosphoryl)phenyl]-9-[(E)-2-(5-methyl-1H-indazol-4-yl)ethenyl]-9H-purin-6-amine', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3ik3', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3ik3_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3ik3', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-08-05', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 513, 'coverage': Decimal('0.254'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.9'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': '0LI', 'identifier_category': 'CCD', 'description': '3-(imidazo[1,2-b]pyridazin-3-ylethynyl)-4-methyl-N-{4-[(4-methylpiperazin-1-yl)methyl]-3-(trifluoromethyl)phenyl}benzam ide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '2qoh', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/2qoh_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/2qoh', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2007-07-20', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.95'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'P3Y', 'identifier_category': 'CCD', 'description': '5-[3-(2-METHOXYPHENYL)-1H-PYRROLO[2,3-B]PYRIDIN-5-YL]-N,N-DIMETHYLPYRIDINE-3-CARBOXAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '2z60', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/2z60_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/2z60', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2007-07-21', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.95'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'P3Y', 'identifier_category': 'CCD', 'description': '5-[3-(2-METHOXYPHENYL)-1H-PYRROLO[2,3-B]PYRIDIN-5-YL]-N,N-DIMETHYLPYRIDINE-3-CARBOXAMIDE', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '3mss', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3mss_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3mss', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2010-04-29', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.95'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B', 'C', 'D']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MS7', 'identifier_category': 'CCD', 'description': 'O-benzyl-N-methyl-L-tyrosinamide', 'chain_ids': ['A', 'B', 'C', 'D']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B', 'C', 'D']}]}}\n",
            "{'summary': {'model_identifier': '3oy3', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3oy3_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3oy3', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2010-09-22', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 511, 'coverage': Decimal('0.252'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.95'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'XY3', 'identifier_category': 'CCD', 'description': '5-[(5-{[4-{[4-(2-hydroxyethyl)piperazin-1-yl]methyl}-3-(trifluoromethyl)phenyl]carbamoyl}-2-methylphenyl)ethynyl]-1-methyl-1H-imidazole-2-carboxamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1abo', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1abo_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1abo', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '1995-05-19', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 61, 'uniprot_end': 121, 'coverage': Decimal('0.054'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.0'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'ABL TYROSINE KINASE', 'chain_ids': ['A', 'B']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P55194', 'identifier_category': 'UNIPROT', 'description': '3BP-1 SYNTHETIC PEPTIDE, 10 RESIDUES', 'chain_ids': ['D', 'C']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SO4', 'identifier_category': 'CCD', 'description': 'SULFATE ION', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3dk7', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3dk7_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3dk7', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2008-06-24', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 233, 'uniprot_end': 505, 'coverage': Decimal('0.243'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.01'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SO4', 'identifier_category': 'CCD', 'description': 'SULFATE ION', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SX7', 'identifier_category': 'CCD', 'description': '2-amino-5-[3-(1-ethyl-1H-pyrazol-5-yl)-1H-pyrrolo[2,3-b]pyridin-5-yl]-N,N-dimethylbenzamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3dk3', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3dk3_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3dk3', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2008-06-24', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 233, 'uniprot_end': 514, 'coverage': Decimal('0.251'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.02'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SO4', 'identifier_category': 'CCD', 'description': 'SULFATE ION', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SX7', 'identifier_category': 'CCD', 'description': '2-amino-5-[3-(1-ethyl-1H-pyrazol-5-yl)-1H-pyrrolo[2,3-b]pyridin-5-yl]-N,N-dimethylbenzamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3dk6', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3dk6_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3dk6', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2008-06-24', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 233, 'uniprot_end': 514, 'coverage': Decimal('0.251'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.02'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SX7', 'identifier_category': 'CCD', 'description': '2-amino-5-[3-(1-ethyl-1H-pyrazol-5-yl)-1H-pyrrolo[2,3-b]pyridin-5-yl]-N,N-dimethylbenzamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '6hd4', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/6hd4_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/6hd4', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2018-08-17', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.03'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'GOL', 'identifier_category': 'CCD', 'description': 'GLYCEROL', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FYW', 'identifier_category': 'CCD', 'description': '6-[(3~{R})-3-oxidanylpyrrolidin-1-yl]-5-pyrimidin-5-yl-~{N}-[4-(trifluoromethyloxy)phenyl]pyridine-3-carboxamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1iep', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1iep_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1iep', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2001-04-10', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.1'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'PROTO-ONCOGENE TYROSINE-PROTEIN KINASE ABL', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3oxz', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3oxz_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3oxz', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2010-09-22', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 511, 'coverage': Decimal('0.252'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.2'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': '0LI', 'identifier_category': 'CCD', 'description': '3-(imidazo[1,2-b]pyridazin-3-ylethynyl)-4-methyl-N-{4-[(4-methylpiperazin-1-yl)methyl]-3-(trifluoromethyl)phenyl}benzam ide', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '6hd6', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/6hd6_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/6hd6', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2018-08-17', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.3'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FYH', 'identifier_category': 'CCD', 'description': '3-(morpholin-4-ylmethyl)-~{N}-[4-(trifluoromethyloxy)phenyl]benzamide', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1fpu', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1fpu_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1fpu', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2000-08-31', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.4'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'PROTO-ONCOGENE TYROSINE-PROTEIN KINASE ABL', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'PRC', 'identifier_category': 'CCD', 'description': 'N-[4-METHYL-3-[[4-(3-PYRIDINYL)-2-PYRIMIDINYL]AMINO]PHENYL]-3-PYRIDINECARBOXAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1m52', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1m52_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1m52', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2002-07-08', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.6'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'PROTO-ONCOGENE TYROSINE-PROTEIN KINASE ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MES', 'identifier_category': 'CCD', 'description': '2-(N-MORPHOLINO)-ETHANESULFONIC ACID', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'P17', 'identifier_category': 'CCD', 'description': '6-(2,6-DICHLORO-PHENYL)-8-METHYL-2-(3-METHYLSULFANYL-PHENYLAMINO)-8H-PYRIDO[2,3-D]PYRIMIDIN-7-ONE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '2hzn', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/2hzn_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/2hzn', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2006-08-09', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.7'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'KIN', 'identifier_category': 'CCD', 'description': '1-[4-(PYRIDIN-4-YLOXY)PHENYL]-3-[3-(TRIFLUOROMETHYL)PHENYL]UREA', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '1abq', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1abq_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1abq', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '1995-05-19', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 61, 'uniprot_end': 121, 'coverage': Decimal('0.054'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.8'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'ABL TYROSINE KINASE SRC-HOMOLOGY 3 (SH3) DOMAIN', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': 'P00520_64-512:1opk.1.A', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://swissmodel.expasy.org/3d-beacons/uniprot/P00520.pdb?range=64-512&template=1opk.1.A&provider=swissmodel', 'model_format': 'PDB', 'model_type': 'ATOMIC', 'model_page_url': 'https://swissmodel.expasy.org/repository/uniprot/P00520?range=64-512&template=1opk.1.A', 'provider': 'SWISS-MODEL', 'created': '2024-03-25', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 64, 'uniprot_end': 512, 'coverage': Decimal('0.4'), 'confidence_type': 'QMEANDisCo', 'confidence_version': '4.3.1', 'confidence_avg_local_score': Decimal('0.805'), 'oligomeric_state': 'MONOMER', 'entities': [{'entity_type': 'NON-POLYMER', 'identifier': 'MYR', 'identifier_category': 'CCD', 'description': 'MYRISTIC ACID', 'chain_ids': ['_']}, {'entity_type': 'NON-POLYMER', 'identifier': 'P16', 'identifier_category': 'CCD', 'description': '6-(2,6-DICHLOROPHENYL)-2-{[3-(HYDROXYMETHYL)PHENYL]AMINO}-8-METHYLPYRIDO[2,3-D]PYRIMIDIN-7(8H)-ONE', 'chain_ids': ['_']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'model based on template 1opk.1.A: Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': 'P00520_121-499:1opl.2.A', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://swissmodel.expasy.org/3d-beacons/uniprot/P00520.pdb?range=121-499&template=1opl.2.A&provider=swissmodel', 'model_format': 'PDB', 'model_type': 'ATOMIC', 'model_page_url': 'https://swissmodel.expasy.org/repository/uniprot/P00520?range=121-499&template=1opl.2.A', 'provider': 'SWISS-MODEL', 'created': '2024-03-25', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 121, 'uniprot_end': 499, 'coverage': Decimal('0.337'), 'confidence_type': 'QMEANDisCo', 'confidence_version': '4.3.1', 'confidence_avg_local_score': Decimal('0.784'), 'oligomeric_state': 'MONOMER', 'entities': [{'entity_type': 'NON-POLYMER', 'identifier': 'P16', 'identifier_category': 'CCD', 'description': '6-(2,6-DICHLOROPHENYL)-2-{[3-(HYDROXYMETHYL)PHENYL]AMINO}-8-METHYLPYRIDO[2,3-D]PYRIMIDIN-7(8H)-ONE', 'chain_ids': ['_']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'model based on template 1opl.2.A: proto-oncogene tyrosine-protein kinase', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': 'P00520_1017-1123:1zzp.1.A', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://swissmodel.expasy.org/3d-beacons/uniprot/P00520.pdb?range=1017-1123&template=1zzp.1.A&provider=swissmodel', 'model_format': 'PDB', 'model_type': 'ATOMIC', 'model_page_url': 'https://swissmodel.expasy.org/repository/uniprot/P00520?range=1017-1123&template=1zzp.1.A', 'provider': 'SWISS-MODEL', 'created': '2024-03-25', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1017, 'uniprot_end': 1123, 'coverage': Decimal('0.095'), 'confidence_type': 'QMEANDisCo', 'confidence_version': '4.3.1', 'confidence_avg_local_score': Decimal('0.78'), 'oligomeric_state': 'MONOMER', 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'model based on template 1zzp.1.A: Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': 'P00520_64-500:1y57.1.A', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://swissmodel.expasy.org/3d-beacons/uniprot/P00520.pdb?range=64-500&template=1y57.1.A&provider=swissmodel', 'model_format': 'PDB', 'model_type': 'ATOMIC', 'model_page_url': 'https://swissmodel.expasy.org/repository/uniprot/P00520?range=64-500&template=1y57.1.A', 'provider': 'SWISS-MODEL', 'created': '2024-03-25', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 64, 'uniprot_end': 500, 'coverage': Decimal('0.389'), 'confidence_type': 'QMEANDisCo', 'confidence_version': '4.3.1', 'confidence_avg_local_score': Decimal('0.707'), 'oligomeric_state': 'MONOMER', 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'model based on template 1y57.1.A: Proto-oncogene tyrosine-protein kinase Src', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': 'AF-P00520-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-P00520-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/P00520', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 1123, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('63.73'), 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': 'P00520', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/P00520', 'model_format': 'MMCIF', 'model_page_url': 'https://alphafill.eu/model?id=P00520', 'provider': 'AlphaFill', 'created': '2023-12-21', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 1123, 'coverage': Decimal('1.0'), 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'description': 'N-(2-CHLORO-6-METHYLPHENYL)-2-({6-[4-(2-HYDROXYETHYL)PIPERAZIN-1-YL]-2-METHYLPYRIMIDIN-4-YL}AMINO)-1,3-THIAZOLE-5-CARBOXAMIDE', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['C', 'S', 'T']}, {'entity_type': 'NON-POLYMER', 'description': '3-(imidazo[1,2-b]pyridazin-3-ylethynyl)-4-methyl-N-{4-[(4-methylpiperazin-1-yl)methyl]-3-(trifluoromethyl)phenyl}benzamide', 'chain_ids': ['D']}, {'entity_type': 'NON-POLYMER', 'description': 'MAGNESIUM ION', 'chain_ids': ['E']}, {'entity_type': 'NON-POLYMER', 'description': 'THIOPHOSPHORIC ACID O-((ADENOSYL-PHOSPHO)PHOSPHO)-S-ACETAMIDYL-DIESTER', 'chain_ids': ['F']}, {'entity_type': 'NON-POLYMER', 'description': '4-[(2,4-dichloro-5-methoxyphenyl)amino]-6-methoxy-7-[3-(4-methylpiperazin-1-yl)propoxy]quinoline-3-carbonitrile', 'chain_ids': ['G']}, {'entity_type': 'NON-POLYMER', 'description': 'CYCLOPROPANECARBOXYLIC ACID {4-[4-(4-METHYL-PIPERAZIN-1-YL)-6-(5-METHYL-2H-PYRAZOL-3-YLAMINO)-PYRIMIDIN-2-YLSULFANYL]-PHENYL}-AMIDE', 'chain_ids': ['H', 'W', 'X', 'Y', 'Z', 'AA', 'AB']}, {'entity_type': 'NON-POLYMER', 'description': '2-METHOXYETHANOL', 'chain_ids': ['I', 'J', 'K']}, {'entity_type': 'NON-POLYMER', 'description': \"ADENOSINE-5'-TRIPHOSPHATE\", 'chain_ids': ['L']}, {'entity_type': 'NON-POLYMER', 'description': \"ADENOSINE-5'-DIPHOSPHATE\", 'chain_ids': ['M']}, {'entity_type': 'NON-POLYMER', 'description': '4-[4-({[3-tert-butyl-1-(quinolin-6-yl)-1H-pyrazol-5-yl]carbamoyl}amino)-3-fluorophenoxy]-N-methylpyridine-2-carboxamide', 'chain_ids': ['N']}, {'entity_type': 'NON-POLYMER', 'description': 'NICKEL (II) ION', 'chain_ids': ['O', 'P']}, {'entity_type': 'NON-POLYMER', 'description': 'SODIUM ION', 'chain_ids': ['Q', 'R', 'U', 'V', 'AP', 'AQ', 'AR', 'AS', 'AT', 'AU', 'AV', 'AW', 'AX', 'AY', 'AZ', 'BA', 'BB', 'BC', 'BD', 'BE', 'BF', 'BG', 'BH', 'BJ', 'BK']}, {'entity_type': 'NON-POLYMER', 'description': '5-AMINO-3-{[4-(AMINOSULFONYL)PHENYL]AMINO}-N-(2,6-DIFLUOROPHENYL)-1H-1,2,4-TRIAZOLE-1-CARBOTHIOAMIDE', 'chain_ids': ['AC']}, {'entity_type': 'NON-POLYMER', 'description': 'ZINC ION', 'chain_ids': ['AD']}, {'entity_type': 'NON-POLYMER', 'description': '1-tert-butyl-3-(naphthalen-1-yl)-1H-pyrazolo[3,4-d]pyrimidin-4-amine', 'chain_ids': ['AE']}, {'entity_type': 'NON-POLYMER', 'description': 'CALCIUM ION', 'chain_ids': ['AF', 'AG', 'AI', 'AN', 'BM']}, {'entity_type': 'NON-POLYMER', 'description': '1-cyclopentyl-3-(1H-pyrrolo[2,3-b]pyridin-5-yl)-1H-pyrazolo[3,4-d]pyrimidin-4-amine', 'chain_ids': ['AH']}, {'entity_type': 'NON-POLYMER', 'description': '1-TERT-BUTYL-3-(4-CHLORO-PHENYL)-1H-PYRAZOLO[3,4-D]PYRIMIDIN-4-YLAMINE', 'chain_ids': ['AJ']}, {'entity_type': 'NON-POLYMER', 'description': 'STAUROSPORINE', 'chain_ids': ['AK']}, {'entity_type': 'NON-POLYMER', 'description': 'ADENOSINE MONOPHOSPHATE', 'chain_ids': ['AL']}, {'entity_type': 'NON-POLYMER', 'description': '(3R)-3-cyclopentyl-3-[4-(7H-pyrrolo[2,3-d]pyrimidin-4-yl)-1H-pyrazol-1-yl]propanenitrile', 'chain_ids': ['AM']}, {'entity_type': 'NON-POLYMER', 'description': '3-CYCLOHEXYL-1-PROPYLSULFONIC ACID', 'chain_ids': ['AO']}, {'entity_type': 'NON-POLYMER', 'description': '1-{(3R)-3-[4-amino-3-(4-phenoxyphenyl)-1H-pyrazolo[3,4-d]pyrimidin-1-yl]piperidin-1-yl}prop-2-en-1-one', 'chain_ids': ['BI']}, {'entity_type': 'NON-POLYMER', 'description': 'S-1,2-PROPANEDIOL', 'chain_ids': ['BL']}]}}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 1.2.&nbsp;  Performing one  filter"
      ],
      "metadata": {
        "id": "jZzlZykLTdnb"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 1.2.1.  Filter by Model type\n",
        "#@markdown  Results can be filtered according to the model category in the 3D-Beacons Network. Models are classified as:\n",
        "\n",
        "#@markdown * EXPERIMENTALLY DETERMINED\n",
        "#@markdown * CONFORMATIONAL ENSEMBLE\n",
        "#@markdown * TEMPLATE-BASED\n",
        "#@markdown * AB-INITIO\n",
        "\n",
        "#@markdown Code below allows you to filter the previous search using a model category.\n",
        "#@markdown\n",
        "model_type = \"EXPERIMENTALLY_DETERMINED\" #@param [\"CONFORMATIONAL_ENSEMBLE\", \"EXPERIMENTALLY_DETERMINED\", \"TEMPLATE-BASED\", \"AB-INITIO\"]\n",
        "model_type_modified= model_type.replace(\"_\", \" \")\n",
        "\n",
        "#Run this code to filter according to model category\n",
        "for structure in structures:\n",
        "    model_category = structure.get('summary', {}).get('model_category')\n",
        "    if model_category == model_type_modified:\n",
        "        print(structure)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LQmlTvt9w392",
        "outputId": "48c1a4c3-8619-4c74-fa5c-e51651f83122",
        "cellView": "form"
      },
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "{'summary': {'model_identifier': '3kfa', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3kfa_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3kfa', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-10-27', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.22'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'B91', 'identifier_category': 'CCD', 'description': '3-{(E)-2-[6-(cyclopropylamino)-9H-purin-9-yl]ethenyl}-4-methyl-N-[3-(4-methyl-1H-imidazol-1-yl)-5-(trifluoromethyl)phenyl]benzamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3k5v', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3k5v_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3k5v', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-10-08', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.74'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STJ', 'identifier_category': 'CCD', 'description': '3-(6-{[4-(trifluoromethoxy)phenyl]amino}pyrimidin-4-yl)benzamide', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1opj', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1opj_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1opj', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2003-03-06', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.75'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MYR', 'identifier_category': 'CCD', 'description': 'MYRISTIC ACID', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1opk', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1opk_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1opk', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2003-03-06', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 27, 'uniprot_end': 515, 'coverage': Decimal('0.435'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.8'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'GOL', 'identifier_category': 'CCD', 'description': 'GLYCEROL', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MYR', 'identifier_category': 'CCD', 'description': 'MYRISTIC ACID', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'P16', 'identifier_category': 'CCD', 'description': '6-(2,6-DICHLOROPHENYL)-2-{[3-(HYDROXYMETHYL)PHENYL]AMINO}-8-METHYLPYRIDO[2,3-D]PYRIMIDIN-7(8H)-ONE', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '3ms9', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3ms9_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3ms9', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2010-04-29', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.8'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MS9', 'identifier_category': 'CCD', 'description': 'methyl 2-amino-4-chlorobenzoate', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '5ih2', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/5ih2_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/5ih2', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2016-02-28', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 757, 'uniprot_end': 765, 'coverage': Decimal('0.008'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.8'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proline rich Peptide', 'chain_ids': ['M', 'N']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'Q64010', 'identifier_category': 'UNIPROT', 'description': 'Adapter molecule crk', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'NA', 'identifier_category': 'CCD', 'description': 'SODIUM ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'PEG', 'identifier_category': 'CCD', 'description': 'DI(HYDROXYETHYL)ETHER', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'P4G', 'identifier_category': 'CCD', 'description': '1-ETHOXY-2-(2-ETHOXYETHOXY)ETHANE', 'chain_ids': ['B', 'N']}]}}\n",
            "{'summary': {'model_identifier': '3kf4', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3kf4_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3kf4', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-10-27', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.9'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'B90', 'identifier_category': 'CCD', 'description': 'N-[4-(dimethylphosphoryl)phenyl]-9-[(E)-2-(5-methyl-1H-indazol-4-yl)ethenyl]-9H-purin-6-amine', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3ik3', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3ik3_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3ik3', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-08-05', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 513, 'coverage': Decimal('0.254'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.9'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': '0LI', 'identifier_category': 'CCD', 'description': '3-(imidazo[1,2-b]pyridazin-3-ylethynyl)-4-methyl-N-{4-[(4-methylpiperazin-1-yl)methyl]-3-(trifluoromethyl)phenyl}benzam ide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '2qoh', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/2qoh_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/2qoh', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2007-07-20', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.95'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'P3Y', 'identifier_category': 'CCD', 'description': '5-[3-(2-METHOXYPHENYL)-1H-PYRROLO[2,3-B]PYRIDIN-5-YL]-N,N-DIMETHYLPYRIDINE-3-CARBOXAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '2z60', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/2z60_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/2z60', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2007-07-21', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.95'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'P3Y', 'identifier_category': 'CCD', 'description': '5-[3-(2-METHOXYPHENYL)-1H-PYRROLO[2,3-B]PYRIDIN-5-YL]-N,N-DIMETHYLPYRIDINE-3-CARBOXAMIDE', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '3mss', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3mss_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3mss', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2010-04-29', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.95'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B', 'C', 'D']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MS7', 'identifier_category': 'CCD', 'description': 'O-benzyl-N-methyl-L-tyrosinamide', 'chain_ids': ['A', 'B', 'C', 'D']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B', 'C', 'D']}]}}\n",
            "{'summary': {'model_identifier': '3oy3', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3oy3_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3oy3', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2010-09-22', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 511, 'coverage': Decimal('0.252'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.95'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'XY3', 'identifier_category': 'CCD', 'description': '5-[(5-{[4-{[4-(2-hydroxyethyl)piperazin-1-yl]methyl}-3-(trifluoromethyl)phenyl]carbamoyl}-2-methylphenyl)ethynyl]-1-methyl-1H-imidazole-2-carboxamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1abo', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1abo_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1abo', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '1995-05-19', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 61, 'uniprot_end': 121, 'coverage': Decimal('0.054'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.0'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'ABL TYROSINE KINASE', 'chain_ids': ['A', 'B']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P55194', 'identifier_category': 'UNIPROT', 'description': '3BP-1 SYNTHETIC PEPTIDE, 10 RESIDUES', 'chain_ids': ['D', 'C']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SO4', 'identifier_category': 'CCD', 'description': 'SULFATE ION', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3dk7', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3dk7_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3dk7', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2008-06-24', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 233, 'uniprot_end': 505, 'coverage': Decimal('0.243'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.01'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SO4', 'identifier_category': 'CCD', 'description': 'SULFATE ION', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SX7', 'identifier_category': 'CCD', 'description': '2-amino-5-[3-(1-ethyl-1H-pyrazol-5-yl)-1H-pyrrolo[2,3-b]pyridin-5-yl]-N,N-dimethylbenzamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3dk3', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3dk3_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3dk3', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2008-06-24', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 233, 'uniprot_end': 514, 'coverage': Decimal('0.251'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.02'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SO4', 'identifier_category': 'CCD', 'description': 'SULFATE ION', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SX7', 'identifier_category': 'CCD', 'description': '2-amino-5-[3-(1-ethyl-1H-pyrazol-5-yl)-1H-pyrrolo[2,3-b]pyridin-5-yl]-N,N-dimethylbenzamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3dk6', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3dk6_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3dk6', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2008-06-24', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 233, 'uniprot_end': 514, 'coverage': Decimal('0.251'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.02'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'SX7', 'identifier_category': 'CCD', 'description': '2-amino-5-[3-(1-ethyl-1H-pyrazol-5-yl)-1H-pyrrolo[2,3-b]pyridin-5-yl]-N,N-dimethylbenzamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '6hd4', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/6hd4_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/6hd4', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2018-08-17', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.03'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'GOL', 'identifier_category': 'CCD', 'description': 'GLYCEROL', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FYW', 'identifier_category': 'CCD', 'description': '6-[(3~{R})-3-oxidanylpyrrolidin-1-yl]-5-pyrimidin-5-yl-~{N}-[4-(trifluoromethyloxy)phenyl]pyridine-3-carboxamide', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1iep', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1iep_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1iep', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2001-04-10', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.1'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'PROTO-ONCOGENE TYROSINE-PROTEIN KINASE ABL', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3oxz', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3oxz_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3oxz', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2010-09-22', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 511, 'coverage': Decimal('0.252'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.2'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': '0LI', 'identifier_category': 'CCD', 'description': '3-(imidazo[1,2-b]pyridazin-3-ylethynyl)-4-methyl-N-{4-[(4-methylpiperazin-1-yl)methyl]-3-(trifluoromethyl)phenyl}benzam ide', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '6hd6', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/6hd6_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/6hd6', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2018-08-17', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.3'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FYH', 'identifier_category': 'CCD', 'description': '3-(morpholin-4-ylmethyl)-~{N}-[4-(trifluoromethyloxy)phenyl]benzamide', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'STI', 'identifier_category': 'CCD', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1fpu', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1fpu_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1fpu', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2000-08-31', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.4'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'PROTO-ONCOGENE TYROSINE-PROTEIN KINASE ABL', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'PRC', 'identifier_category': 'CCD', 'description': 'N-[4-METHYL-3-[[4-(3-PYRIDINYL)-2-PYRIMIDINYL]AMINO]PHENYL]-3-PYRIDINECARBOXAMIDE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '1m52', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1m52_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1m52', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2002-07-08', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.6'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'PROTO-ONCOGENE TYROSINE-PROTEIN KINASE ABL1', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MES', 'identifier_category': 'CCD', 'description': '2-(N-MORPHOLINO)-ETHANESULFONIC ACID', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'P17', 'identifier_category': 'CCD', 'description': '6-(2,6-DICHLORO-PHENYL)-8-METHYL-2-(3-METHYLSULFANYL-PHENYLAMINO)-8H-PYRIDO[2,3-D]PYRIMIDIN-7-ONE', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '2hzn', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/2hzn_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/2hzn', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2006-08-09', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 229, 'uniprot_end': 515, 'coverage': Decimal('0.256'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.7'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'Proto-oncogene tyrosine-protein kinase ABL1', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'KIN', 'identifier_category': 'CCD', 'description': '1-[4-(PYRIDIN-4-YLOXY)PHENYL]-3-[3-(TRIFLUOROMETHYL)PHENYL]UREA', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '1abq', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1abq_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1abq', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '1995-05-19', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 61, 'uniprot_end': 121, 'coverage': Decimal('0.054'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.8'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P00520', 'identifier_category': 'UNIPROT', 'description': 'ABL TYROSINE KINASE SRC-HOMOLOGY 3 (SH3) DOMAIN', 'chain_ids': ['A']}]}}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "cBSLisNPVEoi"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 1.2.2.&nbsp; Filter by Provider\n",
        "\n",
        "#@markdown The 3D-Beacons Network facilitates the aggregation of coordinate files and metadata for both experimental and theoretical protein models. It encompasses a wide range of state-of-the-art and specialized model providers, as well as data from the European Protein Data Bank (PDBe).\n",
        "\n",
        "#@markdown Model providers:\n",
        "#@markdown * PDBe\n",
        "#@markdown * SWISS-MODEL\n",
        "#@markdown * AlphaFold DB\n",
        "#@markdown * Genome3D\n",
        "#@markdown * SASBDB\n",
        "#@markdown * AlphaFill\n",
        "#@markdown * ModelArchive\n",
        "#@markdown * Protein Ensemble Database (PED)\n",
        "\n",
        "#@markdown For further information, visit the [Documentation](https://wwwdev.ebi.ac.uk/pdbe/pdbe-kb/3dbeacons/docs) to check all the current providers.\n",
        "\n",
        "\n",
        "provider = \"AlphaFill\" #@param [\"PDBe\", \"SWISS-MODEL\", \"AlphaFold_DB\", \"Genome3D\", \"SASBDB\", \"AlphaFill\", \"ModelArchive\", \"Protein_Ensemble_Database_(PED)\"]\n",
        "provider_modified= provider.replace(\"_\", \" \")\n",
        "\n",
        "#Run this code to filter results according to given provider\n",
        "for structure in structures:\n",
        "    provider = structure.get('summary', {}).get('provider')\n",
        "\n",
        "    if provider == provider_modified:\n",
        "        print(structure)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q0lLwpB9VWqq",
        "outputId": "b2e0fcae-3970-4f87-9bf4-86b2473e84d1",
        "cellView": "form"
      },
      "execution_count": 18,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "{'summary': {'model_identifier': 'P00520', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/P00520', 'model_format': 'MMCIF', 'model_page_url': 'https://alphafill.eu/model?id=P00520', 'provider': 'AlphaFill', 'created': '2023-12-21', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 1123, 'coverage': Decimal('1.0'), 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'description': 'Tyrosine-protein kinase ABL1', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'description': 'N-(2-CHLORO-6-METHYLPHENYL)-2-({6-[4-(2-HYDROXYETHYL)PIPERAZIN-1-YL]-2-METHYLPYRIMIDIN-4-YL}AMINO)-1,3-THIAZOLE-5-CARBOXAMIDE', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'description': '4-(4-METHYL-PIPERAZIN-1-YLMETHYL)-N-[4-METHYL-3-(4-PYRIDIN-3-YL-PYRIMIDIN-2-YLAMINO)-PHENYL]-BENZAMIDE', 'chain_ids': ['C', 'S', 'T']}, {'entity_type': 'NON-POLYMER', 'description': '3-(imidazo[1,2-b]pyridazin-3-ylethynyl)-4-methyl-N-{4-[(4-methylpiperazin-1-yl)methyl]-3-(trifluoromethyl)phenyl}benzamide', 'chain_ids': ['D']}, {'entity_type': 'NON-POLYMER', 'description': 'MAGNESIUM ION', 'chain_ids': ['E']}, {'entity_type': 'NON-POLYMER', 'description': 'THIOPHOSPHORIC ACID O-((ADENOSYL-PHOSPHO)PHOSPHO)-S-ACETAMIDYL-DIESTER', 'chain_ids': ['F']}, {'entity_type': 'NON-POLYMER', 'description': '4-[(2,4-dichloro-5-methoxyphenyl)amino]-6-methoxy-7-[3-(4-methylpiperazin-1-yl)propoxy]quinoline-3-carbonitrile', 'chain_ids': ['G']}, {'entity_type': 'NON-POLYMER', 'description': 'CYCLOPROPANECARBOXYLIC ACID {4-[4-(4-METHYL-PIPERAZIN-1-YL)-6-(5-METHYL-2H-PYRAZOL-3-YLAMINO)-PYRIMIDIN-2-YLSULFANYL]-PHENYL}-AMIDE', 'chain_ids': ['H', 'W', 'X', 'Y', 'Z', 'AA', 'AB']}, {'entity_type': 'NON-POLYMER', 'description': '2-METHOXYETHANOL', 'chain_ids': ['I', 'J', 'K']}, {'entity_type': 'NON-POLYMER', 'description': \"ADENOSINE-5'-TRIPHOSPHATE\", 'chain_ids': ['L']}, {'entity_type': 'NON-POLYMER', 'description': \"ADENOSINE-5'-DIPHOSPHATE\", 'chain_ids': ['M']}, {'entity_type': 'NON-POLYMER', 'description': '4-[4-({[3-tert-butyl-1-(quinolin-6-yl)-1H-pyrazol-5-yl]carbamoyl}amino)-3-fluorophenoxy]-N-methylpyridine-2-carboxamide', 'chain_ids': ['N']}, {'entity_type': 'NON-POLYMER', 'description': 'NICKEL (II) ION', 'chain_ids': ['O', 'P']}, {'entity_type': 'NON-POLYMER', 'description': 'SODIUM ION', 'chain_ids': ['Q', 'R', 'U', 'V', 'AP', 'AQ', 'AR', 'AS', 'AT', 'AU', 'AV', 'AW', 'AX', 'AY', 'AZ', 'BA', 'BB', 'BC', 'BD', 'BE', 'BF', 'BG', 'BH', 'BJ', 'BK']}, {'entity_type': 'NON-POLYMER', 'description': '5-AMINO-3-{[4-(AMINOSULFONYL)PHENYL]AMINO}-N-(2,6-DIFLUOROPHENYL)-1H-1,2,4-TRIAZOLE-1-CARBOTHIOAMIDE', 'chain_ids': ['AC']}, {'entity_type': 'NON-POLYMER', 'description': 'ZINC ION', 'chain_ids': ['AD']}, {'entity_type': 'NON-POLYMER', 'description': '1-tert-butyl-3-(naphthalen-1-yl)-1H-pyrazolo[3,4-d]pyrimidin-4-amine', 'chain_ids': ['AE']}, {'entity_type': 'NON-POLYMER', 'description': 'CALCIUM ION', 'chain_ids': ['AF', 'AG', 'AI', 'AN', 'BM']}, {'entity_type': 'NON-POLYMER', 'description': '1-cyclopentyl-3-(1H-pyrrolo[2,3-b]pyridin-5-yl)-1H-pyrazolo[3,4-d]pyrimidin-4-amine', 'chain_ids': ['AH']}, {'entity_type': 'NON-POLYMER', 'description': '1-TERT-BUTYL-3-(4-CHLORO-PHENYL)-1H-PYRAZOLO[3,4-D]PYRIMIDIN-4-YLAMINE', 'chain_ids': ['AJ']}, {'entity_type': 'NON-POLYMER', 'description': 'STAUROSPORINE', 'chain_ids': ['AK']}, {'entity_type': 'NON-POLYMER', 'description': 'ADENOSINE MONOPHOSPHATE', 'chain_ids': ['AL']}, {'entity_type': 'NON-POLYMER', 'description': '(3R)-3-cyclopentyl-3-[4-(7H-pyrrolo[2,3-d]pyrimidin-4-yl)-1H-pyrazol-1-yl]propanenitrile', 'chain_ids': ['AM']}, {'entity_type': 'NON-POLYMER', 'description': '3-CYCLOHEXYL-1-PROPYLSULFONIC ACID', 'chain_ids': ['AO']}, {'entity_type': 'NON-POLYMER', 'description': '1-{(3R)-3-[4-amino-3-(4-phenoxyphenyl)-1H-pyrazolo[3,4-d]pyrimidin-1-yl]piperidin-1-yl}prop-2-en-1-one', 'chain_ids': ['BI']}, {'entity_type': 'NON-POLYMER', 'description': 'S-1,2-PROPANEDIOL', 'chain_ids': ['BL']}]}}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 1.3.&nbsp;  Performing two filters\n",
        "\n",
        "This section uses two sets of filters to display the models available in the network"
      ],
      "metadata": {
        "id": "zXmfysj0X25D"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 1.3.1  Filter by Model type and Coverage\n",
        "\n",
        "#markdown The code below will filter all the structures from a Uniprot ID and will save the model with the highest coverage to Google Drive.\n",
        "import urllib.request\n",
        "\n",
        "Uniprot_ID = \"P04637\"  # @param {type:\"string\"}\n",
        "model_type = \"CONFORMATIONAL_ENSEMBLE\"  # @param [\"CONFORMATIONAL_ENSEMBLE\", \"EXPERIMENTALLY_DETERMINED\", \"TEMPLATE-BASED\", \"AB-INITIO\"]\n",
        "model_type_modified = model_type.replace(\"_\", \" \")\n",
        "\n",
        "#search and filter\n",
        "response = urlopen(f\"{WEBSITE_API}{Uniprot_ID}.json\")\n",
        "r = ijson.parse(response)\n",
        "structures = list(ijson.items(r, \"structures.item\", use_float=True))\n",
        "structures.sort(key=lambda x: x.get('summary', {}).get('coverage', 0), reverse=True)\n",
        "\n",
        "highest_coverage_structure = None\n",
        "for structure in structures:\n",
        "    model_category = structure.get('summary', {}).get('model_category')\n",
        "    if model_category == model_type_modified:\n",
        "        highest_coverage_structure = structure\n",
        "        break\n",
        "\n",
        "# Check if a structure was found before accessing its properties\n",
        "if highest_coverage_structure is not None:\n",
        "    print( highest_coverage_structure)\n",
        "\n",
        "else:\n",
        "    print(f\"No structure found with model type: {model_type_modified}\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "TT7vcPdXYRnG",
        "outputId": "56221bec-1a11-4c4a-9a74-ff2083c23e85",
        "cellView": "form"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "{'summary': {'model_identifier': 'PED00438e001', 'model_category': 'CONFORMATIONAL ENSEMBLE', 'model_url': 'https://deposition.proteinensemble.org/api/v1/entries/PED00438/ensembles/e001/ensemble-sample/', 'model_format': 'MMCIF', 'model_page_url': 'https://proteinensemble.org/entries/PED00438', 'provider': 'PED', 'number_of_conformers': Decimal('101.0'), 'ensemble_sample_url': 'https://deposition.proteinensemble.org/api/v1/entries/PED00438/ensembles/e001/ensemble-sample/', 'ensemble_sample_format': 'MMCIF', 'created': '2020-10-01', 'sequence_identity': Decimal('0.9694656488549618'), 'uniprot_start': 1, 'uniprot_end': 393, 'coverage': Decimal('1.0'), 'entities': []}}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 1.3.2&nbsp;  Filter by model type and retrieve results according to resolution.\n",
        "\n",
        "#@markdown The code below will filter all the structures from a Uniprot ID and will save the models with a resolution higher than the specified on the widget.\n",
        "# Run this code to display widgets\n",
        "Uniprot_ID = \"P04637\"  # @param {type:\"string\"}\n",
        "provider = \"PDBe\" #@param [\"PDBe\", \"SWISS-MODEL\", \"AlphaFold_DB\", \"Genome3D\", \"SASBDB\", \"AlphaFill\", \"ModelArchive\", \"Protein_Ensemble_Database_(PED)\"]\n",
        "provider_modified= provider.replace(\"_\", \" \")\n",
        "resolution_search = 1.3 # @param {type:\"slider\", min:0.1, max:5.0, step:0.1}\n",
        "\n",
        "#Run this code to search and filter results according to the input provided\n",
        "r = ijson.parse(urlopen(f\"{WEBSITE_API}{Uniprot_ID}.json\")) #build\n",
        "structures = list(ijson.items(r, \"structures.item\", use_float=True))\n",
        "\n",
        "#filter for experimental structures\n",
        "high_resolution_structures = []\n",
        "for structure in structures:\n",
        "    provider = structure.get('summary', {}).get('provider')\n",
        "    resolution = structure.get('summary', {}).get('resolution')\n",
        "\n",
        "    if provider == provider_modified and resolution is not None and resolution < resolution_search:\n",
        "        # Append the structure to the list without assigning the result back to the list\n",
        "        high_resolution_structures.append(structure)\n",
        "\n",
        "# Perform filter with the high-resolution structures\n",
        "for structure in high_resolution_structures:\n",
        "    print(structure)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "G6L-FpA3lLOt",
        "outputId": "6b64819c-fd33-44c3-8f76-0d25b6201b78",
        "cellView": "form"
      },
      "execution_count": 20,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "{'summary': {'model_identifier': '3d06', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3d06_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3d06', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2008-05-01', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 94, 'uniprot_end': 293, 'coverage': Decimal('0.509'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.2'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'Cellular tumor antigen p53', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'ZN', 'identifier_category': 'CCD', 'description': 'ZINC ION', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '5mhc', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/5mhc_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/5mhc', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2016-11-24', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 382, 'uniprot_end': 393, 'coverage': Decimal('0.031'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.2'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'LYS-LEU-MET-PHE-LYS-TPO-GLU-GLY-PRO-ASP-SER-ASP', 'chain_ids': ['P']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P31947', 'identifier_category': 'UNIPROT', 'description': '14-3-3 protein sigma', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CA', 'identifier_category': 'CCD', 'description': 'CALCIUM ION', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MG', 'identifier_category': 'CCD', 'description': 'MAGNESIUM ION', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '8uqr', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/8uqr_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/8uqr', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-10-24', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 326, 'uniprot_end': 356, 'coverage': Decimal('0.079'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.22'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'Cellular tumor antigen p53', 'chain_ids': ['D', 'A', 'C', 'B']}]}}\n",
            "{'summary': {'model_identifier': '6ggc', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/6ggc_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/6ggc', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2018-05-03', 'sequence_identity': Decimal('98.0'), 'uniprot_start': 94, 'uniprot_end': 312, 'coverage': Decimal('0.557'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.24'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'Cellular tumor antigen p53', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'GOL', 'identifier_category': 'CCD', 'description': 'GLYCEROL', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'ZN', 'identifier_category': 'CCD', 'description': 'ZINC ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'EDO', 'identifier_category': 'CCD', 'description': '1,2-ETHANEDIOL', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'EXN', 'identifier_category': 'CCD', 'description': '[9-ethyl-7-(furan-2-yl)carbazol-3-yl]methyl-methyl-azanium', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '6shz', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/6shz_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/6shz', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2019-08-08', 'sequence_identity': Decimal('98.0'), 'uniprot_start': 94, 'uniprot_end': 311, 'coverage': Decimal('0.555'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.24'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'Cellular tumor antigen p53', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'GOL', 'identifier_category': 'CCD', 'description': 'GLYCEROL', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'ZN', 'identifier_category': 'CCD', 'description': 'ZINC ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'EDO', 'identifier_category': 'CCD', 'description': '1,2-ETHANEDIOL', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '4mzi', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/4mzi_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/4mzi', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2013-09-30', 'sequence_identity': Decimal('92.0'), 'uniprot_start': 93, 'uniprot_end': 292, 'coverage': Decimal('0.509'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.25'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'Cellular tumor antigen p53', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'ZN', 'identifier_category': 'CCD', 'description': 'ZINC ION', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '6gge', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/6gge_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/6gge', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2018-05-03', 'sequence_identity': Decimal('98.0'), 'uniprot_start': 94, 'uniprot_end': 312, 'coverage': Decimal('0.557'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.25000506704'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'Cellular tumor antigen p53', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'ZN', 'identifier_category': 'CCD', 'description': 'ZINC ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'EYE', 'identifier_category': 'CCD', 'description': '[9-ethyl-7-(5-methylthiophen-2-yl)carbazol-3-yl]methyl-methyl-azanium', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '3lw1', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3lw1_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3lw1', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2010-02-23', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 385, 'uniprot_end': 393, 'coverage': Decimal('0.023'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.28'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'peptide of Cellular tumor antigen p53', 'chain_ids': ['P']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P31947', 'identifier_category': 'UNIPROT', 'description': '14-3-3 protein sigma', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'GOL', 'identifier_category': 'CCD', 'description': 'GLYCEROL', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MG', 'identifier_category': 'CCD', 'description': 'MAGNESIUM ION', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '8e7a', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/8e7a_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/8e7a', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-08-23', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 93, 'uniprot_end': 312, 'coverage': Decimal('0.56'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.3'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'Cellular tumor antigen p53', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'ZN', 'identifier_category': 'CCD', 'description': 'ZINC ION', 'chain_ids': ['A']}]}}\n",
            "{'summary': {'model_identifier': '5o1e', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/5o1e_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/5o1e', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2017-05-18', 'sequence_identity': Decimal('98.0'), 'uniprot_start': 94, 'uniprot_end': 312, 'coverage': Decimal('0.557'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.3'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'Cellular tumor antigen p53', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'GOL', 'identifier_category': 'CCD', 'description': 'GLYCEROL', 'chain_ids': ['B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'ZN', 'identifier_category': 'CCD', 'description': 'ZINC ION', 'chain_ids': ['A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': '9GT', 'identifier_category': 'CCD', 'description': '3-iodanyl-2-oxidanyl-5-prop-2-enoxy-4-pyrrol-1-yl-benzoic acid', 'chain_ids': ['A', 'B']}]}}\n",
            "{'summary': {'model_identifier': '6rl3', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/6rl3_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/6rl3', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2019-05-01', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 382, 'uniprot_end': 393, 'coverage': Decimal('0.031'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('1.3'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P04637', 'identifier_category': 'UNIPROT', 'description': 'Cellular tumor antigen p53', 'chain_ids': ['P']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'P31947', 'identifier_category': 'UNIPROT', 'description': '14-3-3 protein sigma', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'GOL', 'identifier_category': 'CCD', 'description': 'GLYCEROL', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'CL', 'identifier_category': 'CCD', 'description': 'CHLORIDE ION', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'MG', 'identifier_category': 'CCD', 'description': 'MAGNESIUM ION', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'K48', 'identifier_category': 'CCD', 'description': '5-azanyl-4-phenyl-thiophene-2-carboximidamide', 'chain_ids': ['A']}]}}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "\n",
        "---\n",
        "\n"
      ],
      "metadata": {
        "id": "409RkUGKZCF3"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "##2.&nbsp; BASIC SEARCH"
      ],
      "metadata": {
        "id": "Q4ccWLiAsqKg"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Perform sequence search\n",
        "\n",
        "#@markdown The 3D-Beacons Network has introduced sequence similarity search functionality which allows you to query the network using the amino acid sequence of a protein.\n",
        "\n",
        "#@markdown The Sequence Similarity Search option available through the network uses the Basic Local Alignment Search Tool (BLAST, Altschul et al., 1990) to find regions of sequence similarity by aligning them with a query sequence. By evaluating the match between the network and query sequence, valuable insights into the structure, function, and evolutionary aspects can be obtained, thus facilitating targeted and systematic exploration of protein structures.\n",
        "\n",
        "#@markdown The code presented below allows you to search the network by performing a sequence-based search via API.\n",
        "\n",
        "sequence = \"MNMLVINGTPRKHGRTRIAASYIAALYHTA\"# @param {type:\"string\"}\n",
        "\n",
        "#Run this code to perform sequence search, it will use the AA from the widget\n",
        "sequence_search(sequence)"
      ],
      "metadata": {
        "id": "QC4XP5UgF0HE",
        "outputId": "9e5d92cc-93e2-4e36-ac43-bde77c5b1b05",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Your search was successful\n",
            "{'accession': 'A0A809FN55', 'id': 'A0A809FN55_BACIU', 'description': 'NAD(P)H-dependent oxidoreductase OS=Bacillus subtilis subsp. subtilis OX=135461 GN=D9C10_03195 PE=3 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('150.0'), 'hsp_bit_score': Decimal('62.3882'), 'hsp_align_len': 29, 'hsp_identity': Decimal('100.0'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_expect': Decimal('6.9E-11')}], 'summary': {'uniprot_entry': {'ac': 'A0A809FN55', 'id': 'A0A809FN55_BACIU', 'uniprot_checksum': 'DCD2A2EAA2677C1E', 'sequence_length': 174, 'segment_start': 1, 'segment_end': 174, 'description': None}, 'structures': [{'summary': {'model_identifier': 'AF-A0A809FN55-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-A0A809FN55-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/A0A809FN55', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.25'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A809FN55', 'identifier_category': 'UNIPROT', 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'A0A809FN55', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/A0A809FN55', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=A0A809FN55', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-15', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E', 'F']}]}}]}, 'hit_uni_ox': 135461, 'hit_uni_os': 'Bacillus subtilis subsp. subtilis', 'hit_com_os': 'Bacillus subtilis subsp. subtilis', 'title': 'NAD(P)H-dependent oxidoreductase'}\n",
            "{'accession': 'A0A0A1M1A3', 'id': 'A0A0A1M1A3_BACIU', 'description': 'FMN-dependent NADPH-azoreductase OS=Bacillus subtilis OX=1423 GN=yhdA PE=3 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('150.0'), 'hsp_bit_score': Decimal('62.3882'), 'hsp_align_len': 29, 'hsp_identity': Decimal('100.0'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_expect': Decimal('6.9E-11')}], 'summary': {'uniprot_entry': {'ac': 'A0A0A1M1A3', 'id': 'A0A0A1M1A3_BACIU', 'uniprot_checksum': 'A51E12FF6D9099D1', 'sequence_length': 174, 'segment_start': 3, 'segment_end': 169, 'description': None}, 'structures': [{'summary': {'model_identifier': 'A0A0A1M1A3_3-169:3gfr.1.B', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://swissmodel.expasy.org/3d-beacons/uniprot/A0A0A1M1A3.pdb?range=3-169&template=3gfr.1.B&provider=swissmodel', 'model_format': 'PDB', 'model_type': 'ATOMIC', 'model_page_url': 'https://swissmodel.expasy.org/repository/uniprot/A0A0A1M1A3?range=3-169&template=3gfr.1.B', 'provider': 'SWISS-MODEL', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2024-02-24', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 3, 'uniprot_end': 169, 'coverage': Decimal('0.96'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'QMEANDisCo', 'confidence_version': '4.3.1', 'confidence_avg_local_score': Decimal('0.9'), 'oligomeric_state': 'HOMODIMER', 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FMN', 'identifier_category': 'CCD', 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['_', '_']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A0A1M1A3', 'identifier_category': 'UNIPROT', 'description': 'model based on template 3gfr.1.A: FMN-dependent NADPH-azoreductase', 'chain_ids': ['A', 'B']}]}}, {'summary': {'model_identifier': 'AF-A0A0A1M1A3-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-A0A0A1M1A3-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/A0A0A1M1A3', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.5'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A0A1M1A3', 'identifier_category': 'UNIPROT', 'description': 'Azoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'A0A0A1M1A3', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/A0A0A1M1A3', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=A0A0A1M1A3', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-13', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'Azoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E']}]}}]}, 'hit_uni_ox': 1423, 'hit_uni_os': 'Bacillus subtilis', 'hit_com_os': 'Bacillus subtilis', 'title': 'FMN-dependent NADPH-azoreductase'}\n",
            "{'accession': 'A0A6M3Z8V9', 'id': 'A0A6M3Z8V9_BACSU', 'description': 'NAD(P)H-dependent oxidoreductase OS=Bacillus subtilis (strain 168) OX=224308 GN=HIR78_05235 PE=3 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('150.0'), 'hsp_bit_score': Decimal('62.3882'), 'hsp_align_len': 29, 'hsp_identity': Decimal('100.0'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_expect': Decimal('6.9E-11')}], 'summary': {'uniprot_entry': {'ac': 'A0A6M3Z8V9', 'id': 'A0A6M3Z8V9_BACSU', 'uniprot_checksum': 'A51E12FF6D9099D1', 'sequence_length': 174, 'segment_start': 3, 'segment_end': 169, 'description': None}, 'structures': [{'summary': {'model_identifier': 'A0A6M3Z8V9_3-169:3gfr.1.B', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://swissmodel.expasy.org/3d-beacons/uniprot/A0A6M3Z8V9.pdb?range=3-169&template=3gfr.1.B&provider=swissmodel', 'model_format': 'PDB', 'model_type': 'ATOMIC', 'model_page_url': 'https://swissmodel.expasy.org/repository/uniprot/A0A6M3Z8V9?range=3-169&template=3gfr.1.B', 'provider': 'SWISS-MODEL', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2024-02-24', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 3, 'uniprot_end': 169, 'coverage': Decimal('0.96'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'QMEANDisCo', 'confidence_version': '4.3.1', 'confidence_avg_local_score': Decimal('0.9'), 'oligomeric_state': 'HOMODIMER', 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FMN', 'identifier_category': 'CCD', 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['_', '_']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A6M3Z8V9', 'identifier_category': 'UNIPROT', 'description': 'model based on template 3gfr.1.A: FMN-dependent NADPH-azoreductase', 'chain_ids': ['A', 'B']}]}}, {'summary': {'model_identifier': 'AF-A0A6M3Z8V9-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-A0A6M3Z8V9-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/A0A6M3Z8V9', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.44'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A6M3Z8V9', 'identifier_category': 'UNIPROT', 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'A0A6M3Z8V9', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/A0A6M3Z8V9', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=A0A6M3Z8V9', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-15', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E', 'F']}]}}]}, 'hit_uni_ox': 224308, 'hit_uni_os': 'Bacillus subtilis (strain 168)', 'hit_com_os': 'Bacillus subtilis (strain 168)', 'title': 'NAD(P)H-dependent oxidoreductase'}\n",
            "{'accession': 'Q8RR37', 'id': 'Q8RR37_GEOSE', 'description': 'Azoreductase OS=Geobacillus stearothermophilus OX=1422 GN=azr PE=2 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('150.0'), 'hsp_bit_score': Decimal('62.3882'), 'hsp_align_len': 29, 'hsp_identity': Decimal('100.0'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_expect': Decimal('6.9E-11')}], 'summary': {'uniprot_entry': {'ac': 'Q8RR37', 'id': 'Q8RR37_GEOSE', 'uniprot_checksum': '0005A2F4AA80EF0A', 'sequence_length': 174, 'segment_start': 1, 'segment_end': 174, 'description': None}, 'structures': [{'summary': {'model_identifier': 'AF-Q8RR37-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-Q8RR37-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/Q8RR37', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.31'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'Q8RR37', 'identifier_category': 'UNIPROT', 'description': 'Azoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'Q8RR37', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/Q8RR37', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=Q8RR37', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-27', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'Azoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E', 'F']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': '~{N}-cyclopropyl-3-oxidanylidene-~{N}-(thiophen-2-ylmethyl)-4~{H}-1,4-benzoxazine-7-carboxamide', 'chain_ids': ['G', 'H']}]}}]}, 'hit_uni_ox': 1422, 'hit_uni_os': 'Geobacillus stearothermophilus', 'hit_com_os': 'Bacillus stearothermophilus', 'title': 'Azoreductase'}\n",
            "{'accession': 'A0A6H0H050', 'id': 'A0A6H0H050_BACIU', 'description': 'NAD(P)H-dependent oxidoreductase OS=Bacillus subtilis subsp. subtilis str. SMY OX=535024 GN=HCN55_05180 PE=3 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('150.0'), 'hsp_bit_score': Decimal('62.3882'), 'hsp_align_len': 29, 'hsp_identity': Decimal('100.0'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_expect': Decimal('6.9E-11')}], 'summary': {'uniprot_entry': {'ac': 'A0A6H0H050', 'id': 'A0A6H0H050_BACIU', 'uniprot_checksum': 'A51E12FF6D9099D1', 'sequence_length': 174, 'segment_start': 3, 'segment_end': 169, 'description': None}, 'structures': [{'summary': {'model_identifier': 'A0A6H0H050_3-169:3gfr.1.B', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://swissmodel.expasy.org/3d-beacons/uniprot/A0A6H0H050.pdb?range=3-169&template=3gfr.1.B&provider=swissmodel', 'model_format': 'PDB', 'model_type': 'ATOMIC', 'model_page_url': 'https://swissmodel.expasy.org/repository/uniprot/A0A6H0H050?range=3-169&template=3gfr.1.B', 'provider': 'SWISS-MODEL', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2024-02-24', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 3, 'uniprot_end': 169, 'coverage': Decimal('0.96'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'QMEANDisCo', 'confidence_version': '4.3.1', 'confidence_avg_local_score': Decimal('0.9'), 'oligomeric_state': 'HOMODIMER', 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FMN', 'identifier_category': 'CCD', 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['_', '_']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A6H0H050', 'identifier_category': 'UNIPROT', 'description': 'model based on template 3gfr.1.A: FMN-dependent NADPH-azoreductase', 'chain_ids': ['A', 'B']}]}}, {'summary': {'model_identifier': 'AF-A0A6H0H050-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-A0A6H0H050-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/A0A6H0H050', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.44'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A6H0H050', 'identifier_category': 'UNIPROT', 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'A0A6H0H050', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/A0A6H0H050', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=A0A6H0H050', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-15', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E', 'F']}]}}]}, 'hit_uni_ox': 535024, 'hit_uni_os': 'Bacillus subtilis subsp. subtilis str. SMY', 'hit_com_os': 'Bacillus subtilis subsp. subtilis str. SMY', 'title': 'NAD(P)H-dependent oxidoreductase'}\n",
            "{'accession': 'O07529', 'id': 'AZR_BACSU', 'description': 'FMN-dependent NADPH-azoreductase OS=Bacillus subtilis (strain 168) OX=224308 GN=azr PE=1 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('150.0'), 'hsp_bit_score': Decimal('62.3882'), 'hsp_align_len': 29, 'hsp_identity': Decimal('100.0'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_expect': Decimal('6.9E-11')}], 'summary': {'uniprot_entry': {'ac': 'O07529', 'id': 'AZR_BACSU', 'uniprot_checksum': 'A51E12FF6D9099D1', 'sequence_length': 174, 'segment_start': 3, 'segment_end': 169, 'description': None}, 'structures': [{'summary': {'model_identifier': '3gfs', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3gfs_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3gfs', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-02-27', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.105'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'O07529', 'identifier_category': 'UNIPROT', 'description': 'FMN-dependent NADPH-azoreductase', 'chain_ids': ['L', 'C', 'G', 'I', 'E', 'J', 'A', 'F', 'H', 'K', 'B', 'D']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FMN', 'identifier_category': 'CCD', 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['L', 'C', 'G', 'I', 'E', 'J', 'A', 'F', 'H', 'K', 'B', 'D']}]}}, {'summary': {'model_identifier': '3gfr', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3gfr_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3gfr', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-02-27', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.403'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'O07529', 'identifier_category': 'UNIPROT', 'description': 'FMN-dependent NADPH-azoreductase', 'chain_ids': ['C', 'D', 'A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FMN', 'identifier_category': 'CCD', 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['C', 'D', 'A', 'B']}]}}, {'summary': {'model_identifier': '1nni', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/1nni_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/1nni', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2003-01-13', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 2, 'uniprot_end': 174, 'coverage': Decimal('0.994'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.5'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'O07529', 'identifier_category': 'UNIPROT', 'description': 'hypothetical protein yhda', 'chain_ids': ['1']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FMN', 'identifier_category': 'CCD', 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['1']}]}}, {'summary': {'model_identifier': '2gsw', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/2gsw_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/2gsw', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2006-04-26', 'sequence_identity': Decimal('100.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.92'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'O07529', 'identifier_category': 'UNIPROT', 'description': 'yhdA', 'chain_ids': ['C', 'D', 'A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FMN', 'identifier_category': 'CCD', 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['C', 'D', 'A', 'B']}]}}, {'summary': {'model_identifier': '3gfq', 'model_category': 'EXPERIMENTALLY DETERMINED', 'model_url': 'https://www.ebi.ac.uk/pdbe/static/entry/3gfq_updated.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://www.ebi.ac.uk/pdbe/entry/pdb/3gfq', 'provider': 'PDBe', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2009-02-27', 'sequence_identity': Decimal('99.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': 'X-RAY DIFFRACTION', 'resolution': Decimal('2.996'), 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'O07529', 'identifier_category': 'UNIPROT', 'description': 'FMN-dependent NADPH-azoreductase', 'chain_ids': ['C', 'D', 'A', 'B']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FMN', 'identifier_category': 'CCD', 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['C', 'D', 'A', 'B']}]}}, {'summary': {'model_identifier': 'O07529_3-169:3gfr.1.B', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://swissmodel.expasy.org/3d-beacons/uniprot/O07529.pdb?range=3-169&template=3gfr.1.B&provider=swissmodel', 'model_format': 'PDB', 'model_type': 'ATOMIC', 'model_page_url': 'https://swissmodel.expasy.org/repository/uniprot/O07529?range=3-169&template=3gfr.1.B', 'provider': 'SWISS-MODEL', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2024-02-24', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 3, 'uniprot_end': 169, 'coverage': Decimal('0.96'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'QMEANDisCo', 'confidence_version': '4.3.1', 'confidence_avg_local_score': Decimal('0.9'), 'oligomeric_state': 'HOMODIMER', 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': 'FMN', 'identifier_category': 'CCD', 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['_', '_']}, {'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'O07529', 'identifier_category': 'UNIPROT', 'description': 'model based on template 3gfr.1.A: FMN-dependent NADPH-azoreductase', 'chain_ids': ['A', 'B']}]}}, {'summary': {'model_identifier': 'AF-O07529-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-O07529-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/O07529', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.66'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'O07529', 'identifier_category': 'UNIPROT', 'description': 'FMN-dependent NADPH-azoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'O07529', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/O07529', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=O07529', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-20', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'FMN-dependent NADPH-azoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E']}]}}]}, 'hit_uni_ox': 224308, 'hit_uni_os': 'Bacillus subtilis (strain 168)', 'hit_com_os': 'Bacillus subtilis (strain 168)', 'title': 'FMN-dependent NADPH-azoreductase'}\n",
            "{'accession': 'E0TX13', 'id': 'E0TX13_BACSH', 'description': 'Oxidoreductase, NAD(P)H-FMN and ferric iron reductase OS=Bacillus spizizenii (strain ATCC 23059 / NRRL B-14472 / W23) OX=655816 GN=yhdA PE=3 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('147.0'), 'hsp_bit_score': Decimal('61.2326'), 'hsp_align_len': 29, 'hsp_identity': Decimal('96.6'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASYISALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAASYI+ALYHT', 'hsp_expect': Decimal('1.9E-10')}], 'summary': {'uniprot_entry': {'ac': 'E0TX13', 'id': 'E0TX13_BACSH', 'uniprot_checksum': 'B55163052D611B7F', 'sequence_length': 174, 'segment_start': 1, 'segment_end': 174, 'description': None}, 'structures': [{'summary': {'model_identifier': 'AF-E0TX13-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-E0TX13-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/E0TX13', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.38'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'E0TX13', 'identifier_category': 'UNIPROT', 'description': 'Oxidoreductase, NAD(P)H-FMN and ferric iron reductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'E0TX13', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/E0TX13', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=E0TX13', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-20', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'Oxidoreductase, NAD(P)H-FMN and ferric iron reductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E']}]}}]}, 'hit_uni_ox': 655816, 'hit_uni_os': 'Bacillus spizizenii (strain ATCC 23059 / NRRL B-14472 / W23)', 'hit_com_os': 'Bacillus subtilis subsp. spizizenii', 'title': 'Oxidoreductase, NAD(P)H-FMN and ferric iron reductase'}\n",
            "{'accession': 'G4NUN2', 'id': 'G4NUN2_BACS4', 'description': 'NADPH azoreductase OS=Bacillus spizizenii (strain DSM 15029 / JCM 12233 / NBRC 101239 / NRRL B-23049 / TU-B-10) OX=1052585 GN=GYO_1222 PE=3 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('147.0'), 'hsp_bit_score': Decimal('61.2326'), 'hsp_align_len': 29, 'hsp_identity': Decimal('96.6'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASYISALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAASYI+ALYHT', 'hsp_expect': Decimal('1.9E-10')}], 'summary': {'uniprot_entry': {'ac': 'G4NUN2', 'id': 'G4NUN2_BACS4', 'uniprot_checksum': 'FEECC781EAD1C3E9', 'sequence_length': 174, 'segment_start': 1, 'segment_end': 174, 'description': None}, 'structures': [{'summary': {'model_identifier': 'AF-G4NUN2-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-G4NUN2-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/G4NUN2', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.44'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'G4NUN2', 'identifier_category': 'UNIPROT', 'description': 'NADPH azoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'G4NUN2', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/G4NUN2', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=G4NUN2', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-20', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'NADPH azoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E', 'F']}]}}]}, 'hit_uni_ox': 1052585, 'hit_uni_os': 'Bacillus spizizenii (strain DSM 15029 / JCM 12233 / NBRC 101239 / NRRL B-23049 / TU-B-10)', 'hit_com_os': 'Bacillus subtilis subsp. spizizenii', 'title': 'NADPH azoreductase'}\n",
            "{'accession': 'Q8RR38', 'id': 'Q8RR38_BACIU', 'description': 'Azoreductase OS=Bacillus subtilis OX=1423 GN=azr PE=2 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('147.0'), 'hsp_bit_score': Decimal('61.2326'), 'hsp_align_len': 29, 'hsp_identity': Decimal('96.6'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASYISALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAASYI+ALYHT', 'hsp_expect': Decimal('1.9E-10')}], 'summary': {'uniprot_entry': {'ac': 'Q8RR38', 'id': 'Q8RR38_BACIU', 'uniprot_checksum': 'B55163052D611B7F', 'sequence_length': 174, 'segment_start': 1, 'segment_end': 174, 'description': None}, 'structures': [{'summary': {'model_identifier': 'AF-Q8RR38-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-Q8RR38-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/Q8RR38', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.5'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'Q8RR38', 'identifier_category': 'UNIPROT', 'description': 'Azoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'Q8RR38', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/Q8RR38', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=Q8RR38', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-27', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'Azoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E']}]}}]}, 'hit_uni_ox': 1423, 'hit_uni_os': 'Bacillus subtilis', 'hit_com_os': 'Bacillus subtilis', 'title': 'Azoreductase'}\n",
            "{'accession': 'A0A6A8FTE9', 'id': 'A0A6A8FTE9_BACIU', 'description': 'NAD(P)H-dependent oxidoreductase OS=Bacillus subtilis OX=1423 GN=GFX43_18650 PE=3 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('144.0'), 'hsp_bit_score': Decimal('60.077'), 'hsp_align_len': 29, 'hsp_identity': Decimal('93.1'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTKIAASYISALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRT+IAASYI+ALYHT', 'hsp_expect': Decimal('5.4E-10')}], 'summary': {'uniprot_entry': {'ac': 'A0A6A8FTE9', 'id': 'A0A6A8FTE9_BACIU', 'uniprot_checksum': 'B5E1D22039718502', 'sequence_length': 174, 'segment_start': 1, 'segment_end': 174, 'description': None}, 'structures': [{'summary': {'model_identifier': 'AF-A0A6A8FTE9-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-A0A6A8FTE9-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/A0A6A8FTE9', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.38'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A6A8FTE9', 'identifier_category': 'UNIPROT', 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'A0A6A8FTE9', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/A0A6A8FTE9', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=A0A6A8FTE9', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-15', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E']}]}}]}, 'hit_uni_ox': 1423, 'hit_uni_os': 'Bacillus subtilis', 'hit_com_os': 'Bacillus subtilis', 'title': 'NAD(P)H-dependent oxidoreductase'}\n",
            "{'accession': 'A0A2M8T5U5', 'id': 'A0A2M8T5U5_9BACI', 'description': 'NAD(P)H-dependent oxidoreductase OS=Bacillus sp. SN1 OX=2055158 GN=CVV77_08285 PE=3 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('142.0'), 'hsp_bit_score': Decimal('59.3066'), 'hsp_align_len': 28, 'hsp_identity': Decimal('96.4'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYH', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASYISALYH', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAASYI+ALYH', 'hsp_expect': Decimal('1.1E-9')}], 'summary': {'uniprot_entry': {'ac': 'A0A2M8T5U5', 'id': 'A0A2M8T5U5_9BACI', 'uniprot_checksum': 'FAB7E31A6ACBEFCB', 'sequence_length': 174, 'segment_start': 1, 'segment_end': 174, 'description': None}, 'structures': [{'summary': {'model_identifier': 'AF-A0A2M8T5U5-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-A0A2M8T5U5-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/A0A2M8T5U5', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.38'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A2M8T5U5', 'identifier_category': 'UNIPROT', 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'A0A2M8T5U5', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/A0A2M8T5U5', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=A0A2M8T5U5', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-14', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E', 'F']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN-ADENINE DINUCLEOTIDE', 'chain_ids': ['G', 'H']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'NADP NICOTINAMIDE-ADENINE-DINUCLEOTIDE PHOSPHATE', 'chain_ids': ['J']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'MENADIONE', 'chain_ids': ['K', 'L', 'M']}]}}]}, 'hit_uni_ox': 2055158, 'hit_uni_os': 'Bacillus sp. SN1', 'hit_com_os': 'Bacillus sp. SN1', 'title': 'NAD(P)H-dependent oxidoreductase'}\n",
            "{'accession': 'D4G6X0', 'id': 'D4G6X0_BACNB', 'description': 'NADPH-dependent FMN reductase-like domain-containing protein OS=Bacillus subtilis subsp. natto (strain BEST195) OX=645657 GN=yhdA PE=3 SV=2', 'hit_length': 172, 'hit_hsps': [{'hsp_score': Decimal('139.0'), 'hsp_bit_score': Decimal('58.151'), 'hsp_align_len': 27, 'hsp_identity': Decimal('100.0'), 'hsp_positive': Decimal('100.0'), 'hsp_qseq': 'MLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_mseq': 'MLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_expect': Decimal('2.8E-9')}], 'summary': {'uniprot_entry': {'ac': 'D4G6X0', 'id': 'D4G6X0_BACNB', 'uniprot_checksum': 'C7D8DBAB0E290E56', 'sequence_length': 172, 'segment_start': 1, 'segment_end': 172, 'description': None}, 'structures': [{'summary': {'model_identifier': 'AF-D4G6X0-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-D4G6X0-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/D4G6X0', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 172, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.25'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'D4G6X0', 'identifier_category': 'UNIPROT', 'description': 'FMN_red domain-containing protein', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'D4G6X0', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/D4G6X0', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=D4G6X0', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-20', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 172, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'FMN_red domain-containing protein', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': '~{N}-cyclopropyl-3-oxidanylidene-~{N}-(thiophen-2-ylmethyl)-4~{H}-1,4-benzoxazine-7-carboxamide', 'chain_ids': ['F', 'G']}]}}]}, 'hit_uni_ox': 645657, 'hit_uni_os': 'Bacillus subtilis subsp. natto (strain BEST195)', 'hit_com_os': 'Bacillus subtilis subsp. natto (strain BEST195)', 'title': 'NADPH-dependent FMN reductase-like domain-containing protein'}\n",
            "{'accession': 'A0A5D4N0H0', 'id': 'A0A5D4N0H0_BACIU', 'description': 'NAD(P)H-dependent oxidoreductase OS=Bacillus subtilis OX=1423 GN=FZC70_19735 PE=3 SV=1', 'hit_length': 174, 'hit_hsps': [{'hsp_score': Decimal('138.0'), 'hsp_bit_score': Decimal('57.7658'), 'hsp_align_len': 29, 'hsp_identity': Decimal('93.1'), 'hsp_positive': Decimal('96.6'), 'hsp_qseq': 'MNMLVINGTPRKHGRTRIAASYIAALYHT', 'hsp_hseq': 'MNMLVINGTPRKHGRTRIAASCISALYHT', 'hsp_mseq': 'MNMLVINGTPRKHGRTRIAAS I+ALYHT', 'hsp_expect': Decimal('4.1E-9')}], 'summary': {'uniprot_entry': {'ac': 'A0A5D4N0H0', 'id': 'A0A5D4N0H0_BACIU', 'uniprot_checksum': '4AB025953F4D86A7', 'sequence_length': 174, 'segment_start': 1, 'segment_end': 174, 'description': None}, 'structures': [{'summary': {'model_identifier': 'AF-A0A5D4N0H0-F1', 'model_category': 'AB-INITIO', 'model_url': 'https://alphafold.ebi.ac.uk/files/AF-A0A5D4N0H0-F1-model_v4.cif', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafold.ebi.ac.uk/entry/A0A5D4N0H0', 'provider': 'AlphaFold DB', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2022-06-01', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': 'pLDDT', 'confidence_version': None, 'confidence_avg_local_score': Decimal('96.44'), 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A5D4N0H0', 'identifier_category': 'UNIPROT', 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}]}}, {'summary': {'model_identifier': 'A0A5D4N0H0', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://alphafill.eu/v1/aff/A0A5D4N0H0', 'model_format': 'MMCIF', 'model_type': None, 'model_page_url': 'https://alphafill.eu/model?id=A0A5D4N0H0', 'provider': 'AlphaFill', 'number_of_conformers': None, 'ensemble_sample_url': None, 'ensemble_sample_format': None, 'created': '2023-12-15', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 174, 'coverage': Decimal('1.0'), 'experimental_method': None, 'resolution': None, 'confidence_type': None, 'confidence_version': None, 'confidence_avg_local_score': None, 'oligomeric_state': None, 'oligomeric_state_confidence': None, 'preferred_assembly_id': None, 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': None, 'identifier_category': None, 'description': 'NAD(P)H-dependent oxidoreductase', 'chain_ids': ['A']}, {'entity_type': 'NON-POLYMER', 'entity_poly_type': None, 'identifier': None, 'identifier_category': None, 'description': 'FLAVIN MONONUCLEOTIDE', 'chain_ids': ['B', 'C', 'D', 'E']}]}}]}, 'hit_uni_ox': 1423, 'hit_uni_os': 'Bacillus subtilis', 'hit_com_os': 'Bacillus subtilis', 'title': 'NAD(P)H-dependent oxidoreductase'}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "\n",
        "---\n",
        "\n"
      ],
      "metadata": {
        "id": "rI73gj5SZKeF"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "##3.&nbsp; RETRIEVING ENSEMBL SUMMARY VIA 3D-BEACONS\n",
        "\n"
      ],
      "metadata": {
        "id": "roqp-1moMR3p"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Retrieve summary using Ensembl Identifier\n",
        "#@markdown The 3D-Beacons network provides the resource for retrieving Ensembl summaries to explore the diverse transcript variants associated with a given gene. ENSEMBL provides information about the different transcript variants for a given gene that map to an UniProt ID.\n",
        "\n",
        "Ensembl_ID_search= \"ENSG00000288864\" # @param {type:\"string\"}\n",
        "\n",
        "ensembl_search(Ensembl_ID_search)"
      ],
      "metadata": {
        "id": "Nad3Gg-fTAyo",
        "outputId": "589b660d-007b-4fd9-8124-e779acb93a62",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form"
      },
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Ensembl search successful\n",
            "{'ensembl_transcript': {'transcript_id': '5507799', 'seqRegionStart': 32613860, 'seqRegionEnd': 32697077, 'alignment_difference': Decimal('0.0')}, 'uniprot_accession': {'uniprot_entry': {'ac': 'A0A8I5KS94', 'id': 'A0A8I5KS94_HUMAN', 'uniprot_checksum': 'EA44F0E7563755ED', 'sequence_length': 623, 'segment_start': 41, 'segment_end': 250, 'description': 'ARHGAP11A-SCG5 readthrough'}, 'structures': [{'summary': {'model_identifier': 'A0A8I5KS94_41-250:3eap.1.A', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://swissmodel.expasy.org/3d-beacons/uniprot/A0A8I5KS94.pdb?range=41-250&template=3eap.1.A&provider=swissmodel', 'model_format': 'PDB', 'model_type': 'ATOMIC', 'model_page_url': 'https://swissmodel.expasy.org/repository/uniprot/A0A8I5KS94?range=41-250&template=3eap.1.A', 'provider': 'SWISS-MODEL', 'created': '2024-02-15', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 41, 'uniprot_end': 250, 'coverage': Decimal('0.337'), 'confidence_type': 'QMEANDisCo', 'confidence_version': '4.3.1', 'confidence_avg_local_score': Decimal('0.761'), 'oligomeric_state': 'MONOMER', 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A8I5KS94', 'identifier_category': 'UNIPROT', 'description': 'model based on template 3eap.1.A: Rho GTPase-activating protein 11A', 'chain_ids': ['A']}]}}]}}\n",
            "{'ensembl_transcript': {'transcript_id': '5507800', 'seqRegionStart': 32615448, 'seqRegionEnd': 32697095, 'alignment_difference': None}, 'uniprot_accession': {'uniprot_entry': {'ac': 'A0A8I5KWH8', 'id': 'A0A8I5KWH8_HUMAN', 'uniprot_checksum': '7AA3847ABA0CC68B', 'sequence_length': 625, 'segment_start': 1, 'segment_end': 252, 'description': 'ARHGAP11A-SCG5 readthrough'}, 'structures': [{'summary': {'model_identifier': 'A0A8I5KWH8_1-252:3eap.1.A', 'model_category': 'TEMPLATE-BASED', 'model_url': 'https://swissmodel.expasy.org/3d-beacons/uniprot/A0A8I5KWH8.pdb?range=1-252&template=3eap.1.A&provider=swissmodel', 'model_format': 'PDB', 'model_type': 'ATOMIC', 'model_page_url': 'https://swissmodel.expasy.org/repository/uniprot/A0A8I5KWH8?range=1-252&template=3eap.1.A', 'provider': 'SWISS-MODEL', 'created': '2024-02-15', 'sequence_identity': Decimal('1.0'), 'uniprot_start': 1, 'uniprot_end': 252, 'coverage': Decimal('0.403'), 'confidence_type': 'QMEANDisCo', 'confidence_version': '4.3.1', 'confidence_avg_local_score': Decimal('0.783'), 'oligomeric_state': 'MONOMER', 'entities': [{'entity_type': 'POLYMER', 'entity_poly_type': 'POLYPEPTIDE(L)', 'identifier': 'A0A8I5KWH8', 'identifier_category': 'UNIPROT', 'description': 'model based on template 3eap.1.A: Rho GTPase-activating protein 11A', 'chain_ids': ['A']}]}}]}}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "\n",
        "---\n",
        "\n"
      ],
      "metadata": {
        "id": "8EDuWyUUZAGh"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "##4.&nbsp; DOWNLOAD SPECIFIC MODEL\n",
        "\n",
        "The downloaded model can be found in the folder \"3D_Beacons_files\" in your Google Drive"
      ],
      "metadata": {
        "id": "f1evRSgcBfs6"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title Download\n",
        "#@markdown Once you found the model/ models you need, you can customise this code to download the MMCIF file to Google Drive\n",
        "\n",
        "#@markdown If multiple models are required, separated them with a comma.\n",
        "Uniprot_ID = \"P04637\"# @param {type:\"string\"}\n",
        "model_download = \"3d06, 4mzi\"  # @param {type:\"string\"}\n",
        "model_download_modified = \", \".join(model_download.strip().split(\",\"))\n",
        "#@markdown The model_download can be found in the key 'model_identifier' within every summary\n",
        "\n",
        "structures = Search3DBeacons(Uniprot_ID)\n",
        "### Run this code to download the models, it will use the input from the widget\n",
        "for structure in structures:\n",
        "  model = structure.get('summary', {}).get('model_identifier')\n",
        "  if model in model_download_modified:\n",
        "    model_url = structure.get(\"summary\", {}).get(\"model_url\")\n",
        "    model_format = structure.get(\"summary\", {}).get(\"model_format\")\n",
        "    file_extension = '.cif' if model_format == 'MMCIF' else '.pdb'\n",
        "    destination_file_path = os.path.join(destination_path, model + file_extension)\n",
        "    download_file(model_url, destination_file_path)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "r_-KL9_9WDQB",
        "outputId": "a1e9ee3c-92ee-4641-8f52-cb5da23198ff",
        "cellView": "form"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "File downloaded successfully to: /content/drive/MyDrive/3D_Beacons_files/3d06.cif\n",
            "File downloaded successfully to: /content/drive/MyDrive/3D_Beacons_files/4mzi.cif\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "\n",
        "---\n",
        "\n"
      ],
      "metadata": {
        "id": "0tSxww15Y-tO"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "# TROUBLESHOOTING\n",
        "\n",
        "| Problem / Response code| Possible cause | Solution |\n",
        "| :---------: |  :- | :- |\n",
        "| 202 | **Accepted request.** <br> 1. The sequence search submission was correct, and the job has been assigned a job identifier. <br> 2. The sequence search job is currently running.| 1.Please wait until the sequence search run completes. It can take 5-10+ minutes.\n",
        "| 400 | **Bad request** <br>  1. Malformed UniProt accession.  <br>  2. Possibly invalid sequence.  <br>  3. Job identifier not found.|1. Please check that the input UniProt accession is correct. <br>  2. Please check your input sequence and retry the submission. <br> 3. Please check if your job identifier is correct.\n",
        "| 404| **Not found.** <br> No results found for the given request.| 1. There may be no results for a specific UniProt accession or protein sequence\n",
        "|500| **Internal Server Error** | 1. This error might be due to scheduled maintenance or rarely technical issues. <br> 2. Please try again later. If the issue persists, please email pdbekb_help@ebi.ac.uk\n",
        "\n"
      ],
      "metadata": {
        "id": "sJYWTEgcU-Na"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Bugs\n",
        "\n",
        "If you encounter any bugs, please report the issue to pdbekb_help@ebi.ac.uk.\n"
      ],
      "metadata": {
        "id": "dANGlNHue_q4"
      }
    }
  ]
}